corporate/lib/stripe.py

```diff
@@ -1553,6 +1553,26 @@ def configure_fixed_price_plan(self, fixed_price: int, sent_invoice_id: str | No
         )
         return f"Customer can now buy a fixed price {required_plan_tier_name} plan."
 
+    def delete_fixed_price_plan(self) -> str:
+        # See configure_fixed_price_plan above for how these CustomerPlan
+        # and CustomerPlanOffer objects are created for fixed-price plans.
+        customer = self.get_customer()
+        assert customer is not None
+        current_plan = get_current_plan_by_customer(customer)
+        if current_plan is None:
+            fixed_price_offer = CustomerPlanOffer.objects.filter(
+                customer=customer, status=CustomerPlanOffer.CONFIGURED
+            ).first()
+            assert fixed_price_offer is not None
+            fixed_price_offer.delete()
+            return "Fixed-price plan offer deleted"
+        fixed_price_next_plan = CustomerPlan.objects.filter(
+            customer=customer, status=CustomerPlan.NEVER_STARTED
+        ).first()
```

Comment:
Is there an added filter we could do that would verify that the next plan we're grabbing is a fixed-price plan, and some other never-started plan?

Reply:
Updated query to:
```python
        fixed_price_next_plan = CustomerPlan.objects.filter(
            customer=customer,
            status=CustomerPlan.NEVER_STARTED,
            fixed_price__isnull=False,
        ).first()
```